### PR TITLE
test: e2e prototype framework stage 1 #199

### LINKED ISSUE #199: [Feature] e2e test
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by hustjieke**
### What type of PR is this?
/test

### Which issue(s) this PR fixes?

Fixes #199 

### What this PR does?

Summary:
1. "make e2e-local" without any other env.
2. here we add module "staging" and use "replace" in go.mod in case we
meet error and fix unknown revision v0.0.0 when use go mod tidy.
see: https://github.com/kubernetes/kubernetes/issues/79384
3. the prototype framework is extracted from kubernetes, for more details,
see: https://github.com/kubernetes/kubernetes/tree/master/test/e2e

4. new file:
test/e2e/e2e.go
test/e2e/framework/config/config.go
test/e2e/framework/test_context.go

[test]
test/e2e/e2e_test.go
test/e2e/framework/config/config_test.go

### CODE REVIEW

**Comment by andyli029** [comment]
Fix the staging.

**Review by andyli029 — COMMENTED** [review]
Fix the staging.

**Comment by hustjieke** [comment]
> Fix the staging.

fixed.

File: go.mod [hunk 1]
```diff
@@ -12,11 +12,17 @@ require (
...
 	github.com/stretchr/testify v1.7.0
 	k8s.io/api v0.21.2
 	k8s.io/apimachinery v0.21.2
 	k8s.io/client-go v0.21.2
-	k8s.io/klog/v2 v2.8.0
+	k8s.io/component-base v0.21.2
+	k8s.io/klog/v2 v2.9.0
 	sigs.k8s.io/controller-runtime v0.9.2
 )
+
```
Comment (andyli029) [inline go.mod:24]: Fix it.

File: go.mod [hunk 2]
```diff
@@ -12,11 +12,17 @@ require (
...
+replace (
```
Comment (andyli029) [inline go.mod:25]: Fix it.

**Comment by andyli029** [comment]
Merge into one commit to the pr

**Review by andyli029 — COMMENTED** [review]
LGTM.

**Pull request merged**